src/evm/EvmStateMachine.ts

```diff
@@ -1,256 +1,208 @@
     }
 
+    /**
+     * Creates a standalone EVM state machine
+     * @param deployStateMachineTx The transaction to deploy the state machine
+     * @param contractInterface The interface of the state machine contract
+     * @returns A new EvmStateMachine instance
+     */
     public static async createStandalone(
         deployStateMachineTx: any,
-        stateMachineInterface: ethers.Interface
+        contractInterface: ethers.Interface
     ): Promise<EvmStateMachine> {
-        let evm = await EVM.create();
+        const evm = await EVM.create();
+
+        // Deploy the state machine contract
         const deploymentResult = await evm.runCall({
             data: ethers.getBytes(deployStateMachineTx.data)
         });
-        if (!deploymentResult.createdAddress)
-            throw new Error(
-                "EvmStateMachine - create - deploymentTx didn't deploy a contract"
-            );
-        return new EvmStateMachine(
+
+        if (deploymentResult.execResult.exceptionError) {
+            throw new Error("EvmStateMachine - create - deploymentTx failed");
+        }
+
+        if (!deploymentResult.createdAddress) {
+            throw new Error("EvmStateMachine - create - deploymentTx didn't deploy a contract");
+        }
+
+        // Create StateMachine with internal ContractExecuter
+        const stateMachine = StateMachine.create(
             evm,
             deploymentResult.createdAddress,
-            stateMachineInterface
+            contractInterface
+        );
+
+
+        return new EvmStateMachine(
+            stateMachine,
+            contractInterface
         );
     }
+
+    /**
+     * Sets up a P2P interaction environment with the state machine
+     * @param signer The signer to use for transactions
+     * @param deployStateMachineTx Transaction to deploy the state machine
+     * @param deployedStateChannelContractInstance The deployed state channel manager proxy
+     * @param stateMachineContractInstance The state machine contract instance
+     * @param p2pEventHooks Optional event hooks for P2P interactions
+     * @returns Promise with the created P2P interaction object
+     */
     public static async p2pSetup<T extends AStateMachineContract>(
         signer: Signer,
         deployStateMachineTx: any,
         deployedStateChannelContractInstance: AStateChannelManagerProxy,
         stateMachineContractInstance: T,
         p2pEventHooks?: P2pEventHooks
     ): Promise<P2pInteraction<T>> {
-        //Sync clock to DLT
+        // Sync clock to DLT
         await Clock.init(signer.provider!);
 
-        deployedStateChannelContractInstance =
-            deployedStateChannelContractInstance.connect(signer);
-        if (DEBUG_CHANNEL_CONTRACT)
+        // Connect signer to state channel contract
+        deployedStateChannelContractInstance = deployedStateChannelContractInstance.connect(signer);
+
+        // Apply debug proxy if enabled
+        const DEBUG_CHANNEL_CONTRACT = true;
```

Comment:
should be false by default

Reply:
Doesn't matter for development you change it to what you need it - it was part of an .env file but removed it since had problems in browsers - however this should be at the top of the file (like in other places) so we don't have to dig to find it 

In production should default to false - maybe instead of a .env file we do a JSON (that works in both nodejs environments and browsers) and for production a build script changes all values to false before building it

Reply:
moved to top. created issue to handle the envs / jsons